app/store/reducers/rackings.js

```diff
@@ -0,0 +1,32 @@
+import { createSlice } from '@reduxjs/toolkit'
+export const slice = createSlice({
+  name: 'rackings',
+  initialState: {
+    loading: false,
+    errorMessage: null,
+    data: {},
+  },
+  reducers: {
+    fetchRackings: (state) => {
+      state.loading = true
+      state.data = {}
+    },
+    fetchRackingsFulfilled: (state, action) => {
+      state.loading = false
+      state.data = action.payload.data
+      state.errorMessage = null
+    },
+    fetchRackingsRejected: (state) => {
+      state.loading = false
+      state.errorMessage = 'Error data list'
```

Comment:
I would update the error message to be more detailed and user friendly. You can then add a selector to use this as an error message to the user.